src/storage/v3/replication/replication_server.cpp

```diff
@@ -167,9 +167,10 @@ void Storage::ReplicationServer::SnapshotHandler(slk::Reader *req_reader, slk::B
       LabelPropertyIndex(&storage_->indices_, &storage_->constraints_, storage_->config_.items);
   try {
     spdlog::debug("Loading snapshot");
-    auto recovered_snapshot = durability::LoadSnapshot(*maybe_snapshot_path, &storage_->vertices_, &storage_->edges_,
-                                                       &storage_->epoch_history_, &storage_->name_id_mapper_,
-                                                       &storage_->edge_count_, storage_->config_.items);
+    auto recovered_snapshot = durability::RecoveredSnapshot{};
+
+    durability::LoadSnapshot(*maybe_snapshot_path, &storage_->vertices_, &storage_->edges_, &storage_->epoch_history_,
+                             &storage_->name_id_mapper_, &storage_->edge_count_, storage_->config_.items);
```

Comment:
Maybe just place TODO to get nack to this, or is there a task for this?

Reply:
I think this should actually work with this changes, because the vertices are still "referenced" by Gid, thus I don't think it requires any TODO for now.